### PR TITLE
Modify the bf16 accuracy checking framework in OpTest

### PR DESCRIPTION
**Pull request by Vvsmile**
### PR types

Others
### PR changes

OPs
### Description

card-72073
- Modify the accuracy checking framework of bf16 datatype in OpTest, including both of forward and backward. 
- The forward result of bf16 is compared with the result of fp32. 
- The backward result of bf16 is compared with the result of fp32. 
- This modification does not need any fix in unit test file. 
- This accuracy checking mechanism is same as it of fp16 now, and this mechanism uses the bf16 or fp16 datatype as the input's datatype, converts to fp32 in the OpTest framework, gets the computation results of each datatype and compares them with each other. Hence, it removes the error caused by rounding from high-precision to low-precision, which is the primary cause resulting in the huge accuracy error in bf16 computation checking in some cases, such as cumsum.
- adjust the tolerance in the backward process of bf16 unit test.
- fix the unique inputs format which causes accuracy checking failed in new framework.
- fix fusion_gru and fusion_lstm which causes the error in the OpTest
- the dequantize is reserved in the white list due to the lacking of fp32 datatype supported in the dequantize operator.

### CODE REVIEW

File: test/legacy_test/eager_op_test.py [hunk 1]
```diff
@@ -550,8 +550,17 @@ def is_fp16_compared_with_fp32(self):
             not in op_accuracy_white_list.NO_FP16_COMPARED_WITH_FP32_OP_LIST
         )
 
+    def is_bf16_compared_with_fp32(self):
+        return self.is_bfloat16_op() and (
+            self.op_type
+            not in op_accuracy_white_list.NO_BF16_COMPARED_WITH_FP32_OP_LIST
```
Comment (ZzSean) [inline test/legacy_test/eager_op_test.py:556]: 这个如果目前没有是不是可以先不加了

File: test/legacy_test/eager_op_test.py [hunk 2]
```diff
@@ -652,46 +661,86 @@ def feed_var(self, input_vars, place):
...
+                                        convert_uint16_to_float(
+                                            np.array(np_value[1])
+                                        )
+                                    )
+                                else:
+                                    tensor.set_recursive_sequence_lengths(
+                                        convert_uint16_to_float(np_value[1])
+                                    )
                             else:
                                 tensor.set_recursive_sequence_lengths(
```
Comment (ZzSean) [inline test/legacy_test/eager_op_test.py:693]: 这个分支还有必要吗

File: test/legacy_test/eager_op_test.py [hunk 3]
```diff
@@ -709,7 +758,7 @@ def _append_ops(self, block):
             self.__class__.use_xpu = True
 
         op_proto = OpProtoHolder.instance().get_op_proto(self.op_type)
-        "infer datatype from inputs and outputs for this test case"
+
```
Comment (ZzSean) [inline test/legacy_test/eager_op_test.py:784]: 可改成注释？无需删除

File: test/legacy_test/eager_op_test.py [hunk 4]
```diff
@@ -550,8 +550,17 @@ def is_fp16_compared_with_fp32(self):
...
+    def is_bf16_compared_with_fp32(self):
+        return self.is_bfloat16_op() and (
+            self.op_type
+            not in op_accuracy_white_list.NO_BF16_COMPARED_WITH_FP32_OP_LIST
+        )
+
     def enable_cal_ref_output(self):
-        self.is_calc_ref = self.is_fp16_compared_with_fp32()
+        self.is_calc_ref = (
+            self.is_fp16_compared_with_fp32()
```
Comment (ZzSean) [inline test/legacy_test/eager_op_test.py:561]: 这个条件用的很多，是不是可以再封装一个函数来把这两个包住，然后这里is_calc_ref的赋值可以简化为直接为True

Comment (Vvsmile) [reply test/legacy_test/eager_op_test.py:561]: Done

File: test/legacy_test/eager_op_test.py (hunk 3 above)
Comment (Vvsmile) [reply test/legacy_test/eager_op_test.py:784]: Done

File: test/legacy_test/eager_op_test.py (hunk 2 above)
Comment (Vvsmile) [reply test/legacy_test/eager_op_test.py:693]: 存在多输入不是FP16和BF16的情况

File: test/legacy_test/eager_op_test.py (hunk 1 above)
Comment (Vvsmile) [reply test/legacy_test/eager_op_test.py:556]: 还包含一个dequantize

File: test/legacy_test/test_unique.py [hunk 5]
```diff
@@ -237,6 +265,34 @@ class TestUniqueOpAxisNoneBF16Op(TestUniqueOpAxisNone):
     def init_dtype(self):
         self.dtype = np.uint16
 
+    def init_config(self):
+        self.inputs = {
+            'X': convert_float_to_uint16(
+                np.random.randint(0, 100, (4, 7, 10)).astype(np.float32)
+            )
+        }
+        unique, indices, inverse, counts = np.unique(
```
Comment (ZzSean) [inline test/legacy_test/test_unique.py]: 简化重复代码

File: test/legacy_test/eager_op_test.py [hunk 6]
```diff
@@ -1754,14 +1827,14 @@ def _compare_numpy(self, name, actual_np, expect_np):
                     ),
                 )
 
-            def _compare_list(self, name, actual, expect):
+                # def _compare_list(self, name, actual, expect):
```
Comment (ZzSean) [inline test/legacy_test/eager_op_test.py]: 这里修改什么意思

File: test/legacy_test/eager_op_test.py (hunk 3 above)
Comment (ZzSean) [reply test/legacy_test/eager_op_test.py:784]: 既然改成注释了就不需要引号了

File: test/legacy_test/eager_op_test.py [hunk 7]
```diff
@@ -550,8 +550,20 @@ def is_fp16_compared_with_fp32(self):
             not in op_accuracy_white_list.NO_FP16_COMPARED_WITH_FP32_OP_LIST
         )
 
+    def is_bf16_compared_with_fp32(self):
+        return self.is_bfloat16_op() and (
+            self.op_type
+            not in op_accuracy_white_list.NO_BF16_COMPARED_WITH_FP32_OP_LIST
+        )
+
+    def is_fp16_or_bf16_compared_with_fp32(self):
```
Comment (ZzSean) [inline test/legacy_test/eager_op_test.py]: 函数名优化一下

**Pull request closed without merging**